packages/eas-cli/src/build/ios/version.ts

```diff
@@ -105,10 +107,17 @@ export async function readShortVersionAsync(
   const workflow = await resolveWorkflowAsync(projectDir, Platform.IOS);
   if (workflow === Workflow.GENERIC) {
     const infoPlist = await readInfoPlistAsync(projectDir, buildSettings);
-    return (
+
+    const shortVersion =
       infoPlist.CFBundleShortVersionString &&
-      evaluateTemplateString(infoPlist.CFBundleShortVersionString, buildSettings)
-    );
+      evaluateTemplateString(infoPlist.CFBundleShortVersionString, buildSettings);
+
+    if (shortVersion && !SHORT_VERSION_REGEX.test(shortVersion)) {
+      throw new Error(
+        `CFBundleShortVersionString must be a period-separated list of at most three non-negative integers. Current value: ${shortVersion}`
+      );
+    }
+    return shortVersion;
   } else {
     return exp.version;
```

Comment:
is this already validated for managed projects? what happens if you specify version in app.json with more than 3 period separated numbers?